compiler/src/codegen/compcore.re

```diff
@@ -1380,8 +1328,52 @@ let compile_array_op = (wasm_mod, env, arr_imm, op) => {
             ),
             get_arr(),
           ),
-          /* [TODO] decref the old item */
-          call_incref(wasm_mod, env, val_),
+          /* [TODO] decref the old item more efficiently (using a swap slot, most likely) */
+          Expression.Tuple_extract.make(
```

Comment:
Primarily because `/* [TODO] decref the old item */` turned unto `decref the old item more efficiently`. We weren't doing the reference management correctly before, but now we are (even if it could be improved in the future).